src/overlays/actors/ovl_En_Dno/z_en_dno.c

```diff
@@ -36,94 +84,949 @@ const ActorInit En_Dno_InitVars = {
     (ActorFunc)EnDno_Draw,
 };
 
-// static ColliderCylinderInit sCylinderInit = {
-static ColliderCylinderInit D_80A73B00 = {
-    { COLTYPE_HIT0, AT_NONE, AC_ON | AC_TYPE_PLAYER, OC1_ON | OC1_TYPE_ALL, OC2_TYPE_1, COLSHAPE_CYLINDER, },
-    { ELEMTYPE_UNK1, { 0x00000000, 0x00, 0x00 }, { 0xF7CFFFFF, 0x00, 0x00 }, TOUCH_NONE | TOUCH_SFX_NORMAL, BUMP_ON, OCELEM_ON, },
+static ColliderCylinderInit sCylinderInit = {
+    {
+        COLTYPE_HIT0,
+        AT_NONE,
+        AC_ON | AC_TYPE_PLAYER,
+        OC1_ON | OC1_TYPE_ALL,
+        OC2_TYPE_1,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK1,
+        { 0x00000000, 0x00, 0x00 },
+        { 0xF7CFFFFF, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_ON,
+        OCELEM_ON,
+    },
     { 17, 58, 0, { 0, 0, 0 } },
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80A73B38[] = {
+static Vec3f D_80A73B2C = { 0.0f, 0.0f, 1.0f };
+
+static InitChainEntry sInitChain[] = {
     ICHAIN_F32(uncullZoneDownward, 80, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_STOP),
 };
 
-#endif
+void func_80A711D0(EnDno* this, GlobalContext* globalCtx, Vec3f* vec) {
+    f32 rand = Rand_ZeroOne() * 0.5f;
+
+    Lights_PointGlowSetInfo(&this->lightInfo, vec->x, vec->y, vec->z, (127.5f * rand) + 127.5f,
+                            (100.0f * rand) + 100.0f, (40.0f * rand) + 40.0f, 320);
+}
+
+s32 func_80A71424(s16* arg0, s16 arg1, s16 yawToPlayer, s16 rotY, s16 arg4, s16 arg5) {
+    s16 temp_v0 = yawToPlayer - rotY;
+    s32 ret;
+
+    if (arg4 >= ABS(temp_v0)) {
+        ret = Math_ScaledStepToS(arg0, arg1 + temp_v0, arg5);
+    } else {
+        ret = Math_ScaledStepToS(arg0, arg1, arg5);
+    }
+
+    return ret;
+}
+
+void func_80A714B4(EnDno* this, GlobalContext* globalCtx) {
+    Actor* actor = NULL;
+
+    do {
```

Comment:
I don't think so in this case. There an unknown loop count due to following the chain of actors, so it should be a while. And then the ending condition is actor being null, which being initialised to null at the start. I can't make it a basic while loop or it'll never run, because actor != null will immediately be false. I can't move the function call out of the loop and also then have it inside the loop, because that just duplicates the function call which isn't right.

I really don't see any way around it here, but if you can whip up some ideas I'm all ears.